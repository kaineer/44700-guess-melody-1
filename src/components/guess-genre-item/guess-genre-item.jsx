import React from 'react';
import {string, number, bool, func} from 'prop-types';
import {AudioPlayer} from '../audio-player/audio-player';

export const GuessGenreItem = ({
  src, orderId, isPlaying, onTogglePlaying
}) => (
  <div className="track">
    <AudioPlayer {...{src, isPlaying, onTogglePlaying}} />

    <div className="game__answer">
      <input className="game__input visually-hidden" type="checkbox" name="answer" value={`answer-${orderId}`} id={`answer-${orderId}`} />
      <label className="game__check" htmlFor={`answer-${orderId}`}>Отметить</label>
    </div>
  </div>
);

GuessGenreItem.propTypes = {
  src: string.isRequired,
  genre: string.isRequired,
  orderId: number.isRequired,
  isPlaying: bool.isRequired,
  onTogglePlaying: func.isRequired
};
