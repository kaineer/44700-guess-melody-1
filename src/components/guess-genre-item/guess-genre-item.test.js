import React from 'react';
import {GuessGenreItem} from './guess-genre-item';
import {create} from 'react-test-renderer';

const src = 'https://foobar.com/1/2/3';
const genre = 'jazz';

describe(`GuessGenreItem`, () => {
  it(`should render correctly`, () => {
    const tree = create(
      <GuessGenreItem
        genre={genre}
        src={src}
        orderId={42}
        key={42}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
