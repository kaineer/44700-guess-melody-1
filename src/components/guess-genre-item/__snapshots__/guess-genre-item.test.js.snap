// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessGenreItem should render correctly 1`] = `
<div
  className="track"
>
  <div
    className="game__track"
  >
    <button
      className="track__button track__button--play"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
    <div
      className="track__status"
    >
      <audio />
    </div>
  </div>
  <div
    className="game__answer"
  >
    <input
      className="game__input visually-hidden"
      id="answer-42"
      name="answer"
      type="checkbox"
      value="answer-42"
    />
    <label
      className="game__check"
      htmlFor="answer-42"
    >
      Отметить
    </label>
  </div>
</div>
`;
